### ISSUE TITLE
Julia REPL freezing VS Code

### ISSUE THREAD

**Issue by CameronBieganek** [JuliaEditorSupport/julia-vscode#517]
My VS Code session freezes periodically when I'm using the built-in Julia REPL. I know this is a rather vague issue, and it might be due to some idiosyncrasy of my system rather than **julia-vscode**, but I don't know where else to look for help. 

**System Info**:
Ubuntu 18.04
Julia 0.6.2
VS Code 1.24.1
VS Code - Julia Language Support 0.10.2

**Comment by davidanthoff** [comment]
I'm just going to call this an upstream bug :) I don't know for sure, but I think in theory we shouldn't even be able to do anything that would lead to this behavior.

**Comment by CameronBieganek** [comment]
Yeah, I found a chunk of code that reliably freezes the REPL (unfortunately it's not a minimal shareable example. I'll see if I can find a minimal example). ...And I just ran it through the integrated REPL in Juno and it freezes Juno too! I also just ran the same code through a stand-alone Julia REPL (via the linux Terminal app) and the stand-alone REPL ran just fine without freezing...

However, note that the VS Code REPL was freezing randomly for other basic code uses, nut just the reliably freezing chunk of code mentioned above. I haven't used Juno enough yet to tell if the REPL freezes on other code.

**Comment by kafisatz** [comment]
I have run into this as well (from time to time)
I am on Julia 0.7alpha though (and maybe it is related to the julia language server not working....)

![image](https://user-images.githubusercontent.com/15026865/41704117-b0c93cbe-7535-11e8-828d-ab00e969355b.png)

**Comment by CameronBieganek** [comment]
I'm not sure if this is related, but I launched VS Code with `code --log trace` and then opened the developer tools by going to Help > Toggle Developer Tools. I observed an error that coincided with some of the freezing episodes:

![vs_code_iswrapped_typeerror](https://user-images.githubusercontent.com/8310743/41739608-34c3cffa-755b-11e8-83d8-a29125f54db2.png)

**Comment by davidanthoff** [comment]
That very much looks like a VS Code issue...

**Comment by CameronBieganek** [comment]
I've noticed the same TypeError in Atom, and someone else has reported it here: https://github.com/JunoLab/atom-ink/issues/164

I should have toggled the stack trace on the javascript error. I'll do that if I ever manage to capture that error again...

**Comment by CameronBieganek** [comment]
Unfortunately I've had a hard time capturing that error again. I don't really seem to have a reliable reproducible way to cause the freezing. However, recently VS Code popped up a message that recommended I switch back to the old DOM integrated terminal because the new one was not performing well on my system, as described [here](https://code.visualstudio.com/updates/v1_24#_integrated-terminal). So far since switching back to the old integrated terminal I haven't had any freezing.

Is xterm.js bundled with VS Code? In other words, will it be updated on my system when the next version of VS Code comes out?

Thanks for the help! Sorry for the non-julia-vscode noise.

**Comment by pfitzseb** [comment]
> Is xterm.js bundled with VS Code? In other words, will it be updated on my system when the next version of VS Code comes out?

Yes, but I'm pretty sure that fix has already been shipped in VSCode. 

It's definitely possible that the GPU rendering was the cause for the crashes.

**Comment by davidanthoff** [comment]
Closing: as noted in the thread, this was an xterm.js problem and the fix shipped in VS Code long ago. Please open a new issue if you see REPL-related freezes on a current release.
